kong/dao/schemas/plugins.lua

```diff
@@ -1,14 +1,31 @@
 local utils = require "kong.tools.utils"
 local Errors = require "kong.dao.errors"
 local db_errors = require "kong.db.errors"
+local singletons = require "kong.singletons"
+local constants = require "kong.constants"
 
 local function load_config_schema(plugin_t)
-  if plugin_t.name then
-    local loaded, plugin_schema = utils.load_module_if_exists("kong.plugins." .. plugin_t.name .. ".schema")
+  local plugin_name = plugin_t.name
+
+  if plugin_name then
+
+    if constants.DEPRECATED_PLUGINS[plugin_name] then
+      ngx.log(ngx.WARN, "the '" .. plugin_name .. "' plugin has been deprecated")
+    end
+
+    -- singletons.configuration would be nil when plugin operations are
+    -- done through DAOs like in migrations or tests
+    if singletons.configuration and not singletons.configuration.plugins[plugin_name] then
```

Comment:
@thibaultcha just want to bring your attention to this condition, it will not be performed in case `singletons` table is not loaded as in case of test we insert plugins before Kong is started.

Do you see any side-effect during runtime?

Reply:
> Do you see any side-effect during runtime?

No I don't, `singletons.configuration` is always initialized in the init phase from the master process